perllib/FixMyStreet/App/Form/Waste/Bulky.pm

```diff
@@ -80,19 +81,57 @@ has_field resident => (
 );
 
 has_field chosen_date => (
-    type => 'Select',
-    widget => 'RadioGroup',
+    type     => 'Select',
+    widget   => 'RadioGroup',
     required => 1,
-    label => 'Available dates',
-    options => [
-        # XXX look these up dynamically
-        { value => '20220825', label => '2022-08-25' },
-        { value => '20220826', label => '2022-08-26' },
-        { value => '20220827', label => '2022-08-27' },
-        { value => '20220828', label => '2022-08-28' },
-    ],
+    label    => 'Available dates',
+    options_method => sub {
```

Comment:
This method seems to get called again for subsequent pages in the process (twice in fact when going to the "add items" page), making everything feel very slow :( I wonder if there's a way to improve that by caching the result, or making sure it's only called on the relevant page, or skipped if the form data already has a `chosen_date` value?

Reply:
I now cache the result, though the duplicate calls / extra calls on subsequent pages ought to be addressed ultimately. I think FormHandler reached a point where it was doing my head in :-(